app/shared/hooks/TextMessagingProvider.js

```diff
@@ -0,0 +1,32 @@
+'use client';
+import { createContext, useState } from 'react';
+import { clientFetch } from 'gpApi/clientFetch';
+import { apiRoutes } from 'gpApi/routes';
+
+export const TextMessagingContext = createContext([{}, () => {}]);
+
+export function TextMessagingProvider({
+  children,
+  textMessaging: initialTextMessaging,
+}) {
+  const [textMessaging, setTextMessaging] = useState(initialTextMessaging);
+
+  const refreshTextMessaging = async () => {
+    try {
+      const resp = await clientFetch(apiRoutes.textMessaging.list);
+
+      setTextMessaging(resp?.status === 404 ? {} : resp.data);
```

Comment:
Should this _only_ be specifically checking for 404? Or should it also be checking for `resp.ok` as well? 🤔